### PR TITLE
CLI and Builder cloud-slang-content dependencies

### PR DESCRIPTION
**Pull request by tethryus**
Automate cloud-slang-content dependencies in the CLI and builder.

Created module cslang-cli-with-content:
- downloads latest content available on cloud-slang-content repository.
  - disabled cache in order to always download the latest content.
- copies all the files and folders needed to work offline.
- downloads all the cs-actions dependencies released on Maven Central. (latest releases).
- leaves the Python packages to be manually installe by the user (can be installed by running pip, in case the user does not have Python installed).

- updated cloudslang-content-verifier to download cs-actions dependencies from cslang-cli-with-content

Package size 94Mb (cslans-cli dependencies now separated from cslang-cli/maven/repo (cs-actions) dependencies).

Signed-off by: Sorin Moldovan <sorin@hpe.com>

### CODE REVIEW

File: cloudslang-cli/pom.xml [hunk 1]
```diff
@@ -323,7 +323,7 @@
                         <phase>generate-resources</phase>
                         <configuration>
                             <target>
-                                <unzip src="${basedir}/src/main/resources/dependency-base.zip" dest="${basedir}/target/cslang-cli/maven/repo">
+                                <unzip src="${basedir}/../cloudslang-content-dependencies/target/dependency-base.zip" dest="${basedir}/target/cslang-cli/maven/repo">
```
Comment (Bonczidai) [inline cloudslang-cli/pom.xml]: into profile - not active by default

File: cloudslang-content-dependencies/pom.xml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                <executions>
+                    <execution>
+                        <id>jar-with-dependencies</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                        <configuration>
+                            <descriptorRefs>
+                                <descriptorRef>jar-with-dependencies</descriptorRef>
```
Comment (Bonczidai) [inline cloudslang-content-dependencies/pom.xml]: do you need this execution?

File: cloudslang-content-dependencies/src/main/assembly/assembly.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<assembly
+    xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2"
```
Comment (Bonczidai) [inline cloudslang-content-dependencies/src/main/assembly/assembly.xml]: maybe move this file under resources

File: cloudslang-content-verifier/pom.xml [hunk 4]
```diff
@@ -278,7 +278,7 @@
                         <phase>generate-resources</phase>
                         <configuration>
                             <target>
-                                <unzip src="${basedir}/../cloudslang-cli/src/main/resources/dependency-base.zip" dest="${basedir}/target/cslang-builder/maven/repo">
+                                <unzip src="${basedir}/../cloudslang-content-dependencies/target/dependency-base.zip" dest="${basedir}/target/cslang-builder/maven/repo">
```
Comment (Bonczidai) [inline cloudslang-content-verifier/pom.xml]: into profile - not active by default

File: pom.xml [hunk 5]
```diff
@@ -758,6 +753,7 @@
         </profile>
     </profiles>
     <modules>
+        <module>cloudslang-content-dependencies</module>
```
Comment (Bonczidai) [inline pom.xml]: IMHO build order is not guaranteed - one solution would be to add an unused declared dependency in the modules that depend on this one

**Comment by tethryus** [comment]
There are still a couple of changes that have to be done in order to make sure everything works as intended.

**Comment by Bonczidai** [comment]
adding @CimpeanRadu

**Comment by Bonczidai** [comment]
@tethryus great stuff here 👍 build manager skill added

- can you please extract a profile for `with-contetent` (`@action` dependencies)
- it would be nice-to-have to include the version numbers in the generated artifact names `cloudslang-cli-with-content-1.0.7-SNAPSHOT.jar`

**Comment by Bonczidai** [comment]
@tethryus did you have a chance to look at the last comment? 
```
can you please extract a profile for with-content (@action dependencies)
it would be nice-to-have to include the version numbers in the generated artifact names cloudslang-cli-with-content-1.0.7-SNAPSHOT.jar
```

**Comment by tethryus** [comment]
I will try and find a good method in which the build process works correctly. Will look into it in the next days.

**Comment by tethryus** [comment]
Closing PR. Other small ones opened that handle the changes in this one.

**Pull request closed without merging**